src/input/mod.rs

```diff
@@ -1449,6 +1449,19 @@ impl State {
             event.time() as u64 * 1000,
         );
 
+        // If the numlock key is pressed ...
+        if event.key_code() == Keycode::new(77) && event.state() == KeyState::Pressed {
```

Comment:
This completely ignores the fact, that we might filter out this event, which would cause the state to mismatch with the actual numlock state.

Instead we should check if the `modifiers_state` has changed after processing keyboard input, e.g. after line 242.

Reply:
Hint: We also don't want to write the file on every potential keyboard event, so we need to compare the old value with the new one, without `numlock_mut()`.